### DISCUSSION TITLE
Problem using CrossSectionExtensions mehtods from MIKE 1D API

### DISCUSSION THREAD

**Discussion by MonicaCoppo** [DHI/Mike1D-SDK-examples#21] [Q&A] [answered]
Hello,

I have been using the Mike1D API trough pythonnet,  to create my code in python for my PhD project. Recently a method that I was using is obsolete (GetNaturalDepthOld). I am trying to use the new equivalent version in the CrossSectionExtensions, CalculateNaturalDepth, but I am having some problems.

I am defining a cross section interface IcsData=ICrossSection(crossSection), for which I want to calculate the natural depth and calling the method as follows for a given Q and slope value:

natural_depth=CrossSectionExtensions.CalculateNaturalDepth(IcsData,float(Q),float(slope))

I am getting and error on de argument definition_
TypeError: No method matches given arguments for CalculateNaturalDepth: (<class DHI.Mike1D.CrossSectionModule.ICrossSection'>, <class 'float'>, <class 'float'>)

Do you have any hint why this is happening or have some examples on using CrossSectionExtensions methods? :)

Thanks in advance,
Best regards,
Monica

**Comment by JesperGr** [comment, marked as answer]
The obsolete method is still working, though it does not take sediments into account, if present. 

The most convenient method is available on the `CrossSectionSlotDecorator`, something like (C# code syntax here):

```
hdParams = new HDParameterData();
csd = new CrossSectionSlotDecorator(ics);
nd = csd.GetNaturalDepth(Q, slope, hdParams);
```

Details of GetNaturalDepth [MIKE 1D Class Library: CrossSectionSlotDecorator.GetNaturalDepth Method](https://manuals.mikepoweredbydhi.help/latest/General/Class_Library/DHI_MIKE1D/html/M_DHI_Mike1D_CrossSectionModule_CrossSectionSlotDecorator_GetNaturalDepth.htm). The `hdParams` are required for some Colebrook-White parameters, in case the cross section resistance is based on that.

It is also possible to use the extension method. Details of the CalculateNaturalDepth can be found here: [MIKE 1D Class Library: CrossSectionExtensions.CalculateNaturalDepth Method](https://manuals.mikepoweredbydhi.help/latest/General/Class_Library/DHI_MIKE1D/html/M_DHI_Mike1D_CrossSectionModule_CrossSectionExtensions_CalculateNaturalDepth.htm). Then it is required to set the `maxDepth` to the height of the cross section. This method will most likely also be deprecated.
